--- a/qttools/src/linguist/linguist.pro
+++ b/qttools/src/linguist/linguist.pro
@@ -4,17 +4,8 @@ requires(qtConfig(linguist))

 TEMPLATE = subdirs
 SUBDIRS  = \
-    lconvert \
-    lprodump \
     lrelease \
-    lrelease-pro \
-    lupdate \
-    lupdate-pro
-!no-png:qtHaveModule(widgets) {
-    QT_FOR_CONFIG += widgets
-    qtConfig(process):qtConfig(pushbutton):qtConfig(toolbutton) {
-        SUBDIRS += linguist
-    }
+    lupdate
 }

 qtNomakeTools( \
--- a/qttools/src/src.pro
+++ b/qttools/src/src.pro
@@ -1,48 +1,6 @@
 TEMPLATE = subdirs

-qtHaveModule(widgets) {
-    no-png {
-        message("Some graphics-related tools are unavailable without PNG support")
-    } else {
-        QT_FOR_CONFIG += widgets
-        qtConfig(pushbutton):qtConfig(toolbutton) {
-            SUBDIRS = assistant \
-                      designer \
-                      pixeltool
-
-            linguist.depends = designer
-        }
-        qtHaveModule(quick):qtConfig(thread):qtConfig(toolbutton): SUBDIRS += distancefieldgenerator
-    }
-}
-
-SUBDIRS += linguist \
-    qtattributionsscanner
-
-qtConfig(library) {
-    !android|android_app: SUBDIRS += qtplugininfo
-}
-
-include($$OUT_PWD/qdoc/qtqdoc-config.pri)
-QT_FOR_CONFIG += qdoc-private
-qtConfig(qdoc): qtConfig(thread): SUBDIRS += qdoc
-
-!android|android_app: SUBDIRS += qtpaths
-
-macos {
-    SUBDIRS += macdeployqt
-}
-
-qtHaveModule(dbus): SUBDIRS += qdbus
-
-win32|winrt:SUBDIRS += windeployqt
-winrt:SUBDIRS += winrtrunner
-qtHaveModule(gui):!wasm:!android:!uikit:!qnx:!winrt: SUBDIRS += qtdiag
-
-qtNomakeTools( \
-    distancefieldgenerator \
-    pixeltool \
-)
+SUBDIRS += linguist

 # This is necessary to avoid a race condition between toolchain.prf
 # invocations in a module-by-module cross-build.
